### ISSUE TITLE
Over-eager rejecting of final private constructors

### ISSUE THREAD

**Issue by Crell** [kalessil/phpinspectionsea#1726]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) or Php Inspections (EA Ultimate), which version |
| Language level | PHP 8.0                        |

Current behaviour
--

The following code gives an error that `final private` isn't allowed.

```php
final private function __construct() {}
```

Expected behaviour
--

`final private` *is* allowed for constructors in PHP 8, precisely because it's the only way to disable constructors entirely.  That was explicitly included in the RFC.  

Environment details
---
IntelliJ Idea 2012.1.1
Plugin version: 4.0.6.3

**Comment by ktomk** [comment]
@Crell Would you be so kind and edit in the exact plugin version and set the language level to PHP 8.0 in your report? Thanks a lot.

**Comment by Crell** [comment]
I don't recall the exact configuration or project I was working in 2 weeks ago, but I added in the plugin version that the Plugin page is showing me right now.

**Comment by ktomk** [comment]
Given

~~~
<?php declare(strict_types=1);

class Foo {
    final private function __construct() { }
}
~~~

I get the inspection message

> Private methods cannot be final as they are never overridden by other classes

at `final` in Phpstorm 2021.1.3 Preview (PS-211.7442.14, May 20) in the PHP 8.0 language level.

However this inspection is displayed _without_ EA inspections (also the message is not prefixed with "[EA]" which I remember being common for the plugin.

@Crell It looks like that this is misplaced and belongs in the Phpstorm tracker (I could yet not figure out the concrete inspection in Phpstorm) and it looks like that it is unreported yet on a first glance. Would you consider to report it in Youtrack and leave a cross reference here?

**Comment by ea-inspections-team** [comment]
Hi folks,

Thanks for reporting @Crell and @ktomk for narrowing down the issue.
Indeed, that's not the extension issue: we are prefixing our messages with '[EA]', so it was easier to find who to reach for support.

Closing the issue.
